More explanatory text.. Fixes #587. Fixes #591 (#602)
tlswg/draft-ietf-tls-esni: draft-ietf-tls-esni.md

achieve this goal without affecting any existing security properties of TLS 1.3.
See {{goals}} for more details about the ECH security and privacy goals.

### Misconfiguration and Deployment Concerns {#misconfiguration}

It is possible for ECH advertisements and servers to become inconsistent. This
may occur, for instance, from DNS misconfiguration, caching issues, or an
incomplete rollout in a multi-server deployment. This may also occur if a server

> loses its ECH keys, or if a deployment of ECH must be rolled back on the server.

Reviewer: ```suggestion
loses its ECH keys, or if a deployment of ECH is rolled back on the server.
```
or "needs to be", but "is" works.

Whatever the cause, there are two main consequences:

* The server has ECH configured but the client has an ECH configuration with a
  key not known to the server.

> * The server has ECH configured but the client has a ECH configuration

Author: ```suggestion
* The server does not have ECH configured but the client has a  ECH configuration
```

>   and so is attempting ECH.

Reviewer 2: Missing a negative in 2nd bullet above?

> * The server has ECH configured but the client has a ECH configuration
>   and so is attempting ECH.

Reviewer: ```suggestion
* The server does not have ECH configured but the client has a ECH configuration
  and so is attempting ECH.
```

inconsistencies between systems that advertise ECH keys and servers, at the cost
of extra round-trips due to a retry. Two specific scenarios are detailed below.

### Misconfiguration and Deployment Concerns {#misconfiguration-deployment}

The retry mechanism described in {{rejected-ech}} allows the server to
recover from situations where it is not able to decrypt the ECH field,
for instance if the client has a stale ECHConfig or the server has lost

> the keys. The security of this mechanism is rooted in the server having

Reviewer: ```suggestion
the keys. The security of this mechanism depends on the server having
```

a certificate for the public name in the ECHConfig.

Unless ECH is disabled as a result of successfully establishing a
connection to the public name, the client MUST NOT fall back to using
unencrypted ClientHellos. This means that there are situations where
the client may not be able to connect to the server at all, such as:

- If there is a transport-level failure
- If TLS handshake does not complete successfully
- If TLS handshake completes using the ClientHelloOuter but with a certificate
  which is not valid for the public name.

> Falling back to unencrypted ClientHellos in any of these cases would

Reviewer: ```suggestion
Falling back to an unencrypted ClientHello in any of these cases would
```

allow a network attacker to disclose the contents of this ClientHello,
including the SNI. This restriction may mean that certain kinds of
network misbehavior or misconfiguration (see {{middleboxes}} may make
it impossible to complete an ECH connection where an ordinary TLS 1.3
connection would work.

Clients MAY attempt to use another server or ECHConfig value from the
DNS results, if one is provided.

> 

Reviewer 3: Looks spurious

> 

Author: ```suggestion
```
